pallets/asset-index/src/lib.rs

```diff
@@ -673,6 +713,51 @@ pub mod pallet {
 			share.checked_div(&nav).ok_or_else(|| ArithmeticError::Overflow.into())
 		}
 
+		/// Consolidate and clean deposits while completing withdraw
+		fn do_consolidate_deposits(caller: &T::AccountId, mut amount: T::Balance) -> Result<T::Balance, DispatchError> {
+			let withdrawal_amount = amount.clone();
+			<Deposits<T>>::try_mutate_exists(&caller, |maybe_depositing| -> Result<T::Balance, DispatchError> {
+				let depositing = maybe_depositing.take().ok_or(<Error<T>>::NoDeposits)?;
+				let mut start: T::BlockNumber = 0_u32.into();
+
+				let still_depositing: Vec<_> = depositing
```

Comment:
`deposits[0]` is always the oldest and we only take until `amount` is reached. 
can we youse `deposits.retain` here instead?